src/shared/messaging/test/port-rpc-test.js

```diff
@@ -175,23 +175,158 @@ describe('PortRPC', () => {
     assert.calledWith(testMethod, 'second', 'call');
   });
 
-  describe('#disconnect', () => {
-    it('disconnects and returns port', async () => {
-      const rpc = new PortRPC();
-      const testMethod = sinon.stub();
-      rpc.on('test', testMethod);
+  it('should send "connect" event after connection', async () => {
+    const { port1, port2 } = new MessageChannel();
+    const sender = new PortRPC();
+    const receiver = new PortRPC();
+    const connectHandler = sinon.stub();
+
+    receiver.on('connect', connectHandler);
+    receiver.connect(port2);
+    sender.connect(port1);
+
+    await waitForMessageDelivery();
+
+    assert.calledWith(connectHandler);
+  });
+
+  it('should send "close" event after port is destroyed', async () => {
+    const { port1, port2 } = new MessageChannel();
+    const sender = new PortRPC();
+    const receiver = new PortRPC();
+    const closeHandler = sinon.stub();
+
+    receiver.on('close', closeHandler);
+    receiver.connect(port2);
+    sender.connect(port1);
+    await waitForMessageDelivery();
+
+    closeHandler.resetHistory();
+    sender.destroy();
+    await waitForMessageDelivery();
+
+    assert.calledWith(closeHandler);
+  });
+
+  it('should send "close" event when window is unloaded', async () => {
+    const { port1, port2 } = new MessageChannel();
+    const sender = new PortRPC();
+    const receiver = new PortRPC();
+    const closeHandler = sinon.stub();
 
-      assert.isNull(rpc.disconnect());
+    receiver.on('close', closeHandler);
+    receiver.connect(port2);
+    sender.connect(port1);
+    await waitForMessageDelivery();
 
-      const { port1 } = new MessageChannel();
-      rpc.connect(port1);
+    closeHandler.resetHistory();
```

Comment:
```suggestion
```

Reply:
I changed this to an assert. See https://github.com/hypothesis/client/pull/4175#discussion_r801355241.